bin/ruby-build

```diff
@@ -1081,6 +1081,16 @@ has_broken_mac_openssl() {
   [[ "$RUBY_CONFIGURE_OPTS" != *--with-openssl-dir=* ]]
 }
 
+has_openssl_3() {
+  local openssl_version="$(openssl version 2>/dev/null || printf '#include <openssl/opensslv.h>\nOPENSSL_VERSION_TEXT\n' | cc -xc -E - 2>/dev/null | tail -n1 | tr -d '"' || true)"
```

Comment:
I don't think it's necessary to do so much output sanitization with `tail | tr`. Just pipe the version output to `grep -q "OpenSSL 3"` and see if it matches.